com.github.jeremypw.gnonograms.json

```diff
@@ -0,0 +1,45 @@
+{
+    "app-id": "com.github.jeremypw.gnonograms",
+    "base": "io.elementary.BaseApp",
+    "base-version": "juno-19.08",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "19.08",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "com.github.jeremypw.gnonograms",
+
+    "finish-args": [
+      "--share=ipc",
+      "--socket=fallback-x11",
+      "--socket=wayland",
+      "--filesystem=home",
```

Comment:
Does Gnonograms really needs access to the entire user's home? Or could it be only a subset within that location?

Reply:
The program saves unfinished games in a temporary folder under `.local\share`.  The user can also save named games, e.g. ones they have designed and also load named from the filesystem but there isn't a fixed place set by the code - that is up to the user. I wasn't sure how much to restrict this.

Reply:
If you're using a GTKFileChooserNative then that should help you negotiate the portal without having to have the whole system available. https://developer.gnome.org/gtk3/stable/gtk3-GtkFileChooserNative.html worth giving that a quick try if you can.

Aside from that I tested using `flatpak run --nofilesystem=home com.github.jeremypw.gnonograms` and the in progress game is indeed saved within the sandbox, so you don't need the whole filesystem for that.

Reply:
OK, I'll try and tighten up the filesystem sandboxing.  I assumed things saved within the sandbox were lost when the sandbox was closed. However, there is still the question of how to provide the possibility of loading games designed externally from the host filesystem.  I guess using GtkFileChooserNative should deal with that.

Reply:
There are several limitation to the FileChooserNative portal, though.  You cannot add extra widgets or actions or any file filters.  So some things like giving a choice whether to save the game with or without the solution by means of a switch within the dialog will have to be rethought.

Reply:
> There are several limitation to the FileChooserNative portal, though. You cannot add extra widgets or actions or any file filters. So some things like giving a choice whether to save the game with or without the solution by means of a switch within the dialog will have to be rethought.

You can set a filter and add actions to the FileChooserNative. You can't though add widgets because it could be a Qt portal if you're using KDE for example.